catroid/src/main/java/org/catrobat/catroid/formulaeditor/datacontainer/SpriteVariableBehaviour.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2017 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.formulaeditor.datacontainer;
+
+import org.catrobat.catroid.content.Scene;
+import org.catrobat.catroid.content.Sprite;
+import org.catrobat.catroid.formulaeditor.UserVariable;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+class SpriteVariableBehaviour extends SpriteDataBehaviour<Sprite, UserVariable> {
+
+	private DataContainer dataContainer;
+
+	SpriteVariableBehaviour(DataContainer dataContainer) {
+		this.dataContainer = dataContainer;
+	}
+
+	@Override
+	public List<UserVariable> getOrCreate(Sprite key) {
+		List<UserVariable> data = super.getOrCreate(key);
+		removeIllegalSpriteVariableEntries(key);
+		return data;
+	}
+
+	@Override
+	Map<Sprite, List<UserVariable>> cloneForScene(Scene scene, Map<Sprite, List<UserVariable>> originalData) {
+		removeIllegalSpriteVariableEntries(null);
```

Comment:
I really don't like this kind of bug workarounds, cause things like this leave a lot of unused code when the actual bug is fixed.

IMO since you're editing the module in which the bug appears it should be fixed right away.